src/Rings/mpoly-ideals.jl

```diff
@@ -180,7 +180,10 @@ function Base.:*(I::MPolyIdeal{T}, J::MPolyIdeal{T}) where T
       end
     end
   end
-  return ideal(base_ring(I), unique!(newgens))
+  result = ideal(base_ring(I), unique!(newgens))
+  # carry over eventual information about the dimension
+  is_known(dim, I) && is_known(dim, J) && (result.dim = maximum([dim(I), dim(J)]))
```

Comment:
we should use short-circuiting only for control flow, not like here for other side effects such as assignments (this is now explicitly pointed out in our [developer docs](https://docs.oscar-system.org/dev/DeveloperDocumentation/styleguide/#Code-structure), I think @thofma added it there recently -- and I agree).

So:
```suggestion
  if is_known(dim, I) && is_known(dim, J)
    result.dim = maximum([dim(I), dim(J)])
  end
```
And while we are a it: don't create a vector just to find the larger of two numbers, just use `max`:
```suggestion
  if is_known(dim, I) && is_known(dim, J)
    result.dim = max(dim(I), dim(J))
  end
```